jbpm-case-mgmt/jbpm-case-mgmt-api/src/main/java/org/jbpm/casemgmt/api/event/CaseEventListener.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.casemgmt.api.event;
+
+import java.util.EventListener;
+
+/**
+ * Case event listener that is invoked upon various
+ * operation related to a case.
+ *
+ */
+public interface CaseEventListener extends EventListener {
+
+    void beforeCaseStarted(StartCaseEvent event);
+
+    void afterCaseStarted(StartCaseEvent event);
+
+    void beforeCaseCancelled(CancelCaseEvent event);
+    
+    void afterCaseCancelled(CancelCaseEvent event);
+
+    void beforeCaseDestroyed(CaseEvent event);
```

Comment:
It's probably worthwhile to create a specific event to handle the case destroy call, perhaps CaseDestroyEvent?